esphome/components/atm90e32/atm90e32.cpp

```diff
@@ -156,19 +189,18 @@ void ATM90E32Component::dump_config() {
   LOG_SENSOR("  ", "Frequency", this->freq_sensor_);
   LOG_SENSOR("  ", "Chip Temp", this->chip_temperature_sensor_);
 }
-float ATM90E32Component::get_setup_priority() const { return setup_priority::DATA; }
+
+float ATM90E32Component::get_setup_priority() const { return setup_priority::BUS; }
```

Comment:
This should be `IO` at the earliest as it relies on the spi bus to be setup.
```suggestion
float ATM90E32Component::get_setup_priority() const { return setup_priority::IO; }
```

Reply:
Ok, But I think the priority description should change then. It's not clear to me, do we set the priority of the component to a value lower than the interfaces it will use or at them?

Here the description is misleading unless we indicate that we need to be lower than i2c or spi when we depend on a BUS.

e.g.
const float esphome::setup_priority::BUS
For communication buses like i2c/spi

Reply:
Lower number = lower priority = setup later

Reply:
If a component is set at the same priority level, it could be setup out of order

Reply:
Ok, I'm a bit confused by this .. we init the SPI interface in this component so wouldn't this require a higher priority.

e.g. from setup the first thing we call is:
  this->spi_setup(); and there is and option to have multiple instances in the configuration.

Reply:
I think I see it now, SPI is a separate component and has it's own priority setting.

Reply:
Yes, the SPI base component needs to do its own periperhal setup before any components try to register themselves with it.